pkg/build/generator/generator.go

```diff
@@ -258,24 +288,57 @@ func (g *BuildGenerator) resolveImageRepoReference(ctx kapi.Context, from *kapi.
 	} else {
 		namespace = defaultNamespace
 	}
-	imageStream, err := g.Client.GetImageStream(kapi.WithNamespace(ctx, namespace), from.Name)
-	if err != nil {
-		if errors.IsNotFound(err) {
+
+	switch from.Kind {
+	case "ImageStreamImage":
+		bits := strings.Split(from.Name, "@")
+		stream, err := g.Client.GetImageStream(kapi.WithNamespace(ctx, namespace), bits[0])
```

Comment:
It might be easier to get the ImageStreamImage using from.Name, and use its DockerImageReference. I think that would be cleaner and give the same result.